app/javascript/react/components/MapButtons/MapButtons.tsx

```diff
@@ -106,21 +114,19 @@ const MapButtons = () => {
             <MapButton
               title={t("navbar.copyLink")}
               image={copyLinkIcon}
-              onClick={() => {}}
+              onClick={() => handleCopyLinkClick()}
               alt={t("navbar.altCopyLink")}
-              isActive={activeButtons.includes(ButtonTypes.COPY_LINK)}
+              isActive={activeCopyLinkButton}
               className="active-overlay"
             />
           }
           isIconOnly={false}
           showBelowButton
           onOpen={() => {
-            handleClick(ButtonTypes.COPY_LINK);
+            handleCopyLinkClick();
           }}
           onClose={() => {
-            setActiveButtons((prevState) =>
-              prevState.filter((type) => type !== ButtonTypes.COPY_LINK)
-            );
+            setActiveCopyLinkButton(false);
```

Comment:
We can also unify this logic and have open/close in `handleCopyLinkClick()` similar to `handleTimelapseClick()`